### PR TITLE
Updating to Capistrano 3

### PR DESCRIPTION
**Pull request by ortegacmanuel**
- Added capistrano configuration files
- Loaded figaro env variables into capistrano proccees
- Added deploy variables to application.yml file
- App generator now creates a secret key in order be able to create the example app
- Added capistrano task for uploading linked files [application.yml, database.yml and secrets.yml]

### CODE REVIEW

File: lib/generators/templates/Capfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Load ENV vars via Figaro
+require 'figaro'
+Figaro.application = Figaro::Application.new(environment: 'production', path: File.expand_path('../config/application.yml', __FILE__))
```
Comment (acesuares) [inline lib/generators/templates/Capfile]: Why production?

File: lib/generators/templates/application.yml.blank [hunk 2]
```diff
@@ -3,6 +3,8 @@ SECRET_TOKEN:
 
 DEPLOY_USER:
 DEPLOY_HOST:
+DEPLOY_APPLICATION:
+DEPLO_REPO_URL:
```
Comment (acesuares) [inline lib/generators/templates/application.yml.blank]: typo? should  be deploy not deplo?

File: lib/generators/templates/deploy.rb [hunk 3]
```diff
@@ -1,82 +1,39 @@
-# -*- encoding : utf-8 -*-
+set :application, ENV['DEPLOY_APPLICATION']
+set :repo_url, ENV['DEPLO_REPO_URL']
```
Comment (acesuares) [inline lib/generators/templates/deploy.rb]: same typo here

File: lib/generators/templates/application.yml.blank (hunk 2 above)
Comment (ortegacmanuel) [reply lib/generators/templates/application.yml.blank]: Yes! :( should be deploy!

File: lib/generators/templates/deploy.rb (hunk 3 above)
Comment (ortegacmanuel) [reply lib/generators/templates/deploy.rb]: Yes! copy & paste :(

File: lib/generators/templates/Capfile (hunk 1 above)
Comment (ortegacmanuel) [reply lib/generators/templates/Capfile]: the reason behind `production` is that for deployment we only need the production variables values and the global ones.

**Pull request merged**